eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/ConnectionTlv.scala

```diff
@@ -42,8 +43,32 @@ object InitTlvCodecs {
 
   private val networks: Codec[Networks] = variableSizeBytesLong(varintoverflow, list(bytes32)).as[Networks]
 
-  val initTlvCodec = TlvCodecs.tlvStream(discriminated[InitTlv].by(varint)
+  val initTlvCodec = tlvStream(discriminated[InitTlv].by(varint)
     .typecase(UInt64(1), networks)
   )
 
+}
+
+sealed trait WarningTlv extends Tlv
```

Comment:
Nit: this file could have been named `SetupAndControlTlv.scala`.

One day we may split the `LightningMessageCodecs` file between `SetupAndControl`, `Channel`, `Commitment`, `Routing`, etc.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1891/commits/003c0db9cc9c6c1263fa5639bdb58c61e7e7641e